src/main/java/Duke.java

```diff
@@ -1,10 +1,322 @@
+                        }
+                        if(list.isEmpty()) {
+                            throw new DukeException("☹ OOPS!!! There are no tasks to be marked as done.");
+                        }
+                        if(Integer.parseInt(array[1]) > list.size()) {
+                            throw new DukeException("☹ OOPS!!! There is no such task to be marked as done.");
+                        } else {
+                            doneCommand(list, array);
+                        }
+                        total--;
+                    } else if (s.contains("todo")) {
+                        if (s.length() <= 5) {
+                            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                        }
+                        String st = s.substring(5);
+                        toDoCommand(st, list);
+                        total++;
+                        continue;
+                    } else if (s.contains("deadline")) {
+                        if (s.length() <= 9 || !s.contains("/by")) {
+                            throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                        }
+                        String[] strArr = s.split("/by ");
+                        String description = strArr[0].substring(9).trim();
+                        String by = strArr[1];
+                        deadlineCommand(description, by, list);
+                        total++;
+                        continue;
+
+                    } else if (s.contains("event")) {
+                        if (s.length() <= 6 || !s.contains("/at")) {
+                            throw new DukeException("☹ OOPS!!! The description of an event cannot be empty.");
+                        }
+                        String[] strArr = s.split("/at ");
+                        String description = strArr[0].substring(6).trim();
+                        String date = strArr[1];
+                        eventCommand(description, date, list);
+                        total++;
+                        continue;
+                    } else if (s.contains("delete")) {
+                        String[] strArr = s.split(" ");
+                        if(strArr.length == 1) {
+                            throw new DukeException("☹ OOPS!!! I don't know which task to delete.");
+                        }
+                        int idx = Integer.parseInt(strArr[1]) - 1;
+                        deleteCommand(idx, list);
+                        total--;
+                    } else {
+                        throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                    }
+                } catch (DukeException ex) {
+                    System.out.println(ex);
+                }
+            }
+        }
+     }
+
+    /**
+     * prints horizontal line
+     */
```

Comment:
The function name is rather self-explanatory. Could the javadoc be removed without losing code quality?